### ISSUE TITLE
Netatmo Binding: Nullpointer Exception

### ISSUE THREAD

**Issue by juelicher** [openhab/openhab#1820]
I found the following exception in my log file
2014-12-10 18:25:46.646 ERROR o.o.c.s.AbstractActiveService[:175]- Error while executing background thread Netatmo Refresh Service
java.lang.NullPointerException: null
	at org.openhab.binding.netatmo.internal.NetatmoBinding.execute(NetatmoBinding.java:120)
	at org.openhab.core.binding.AbstractActiveBinding$BindingActiveService.execute(AbstractActiveBinding.java:156)
	at org.openhab.core.service.AbstractActiveService$RefreshThread.run(AbstractActiveService.java:173)

Immediately before that was an connection error, but I am not sure if these errors are related:
2014-12-10 18:25:46.097 WARN  o.openhab.io.net.http.HttpUtil[:205]- Method failed: HTTP/1.1 403 Forbidden

**Comment by juelicher** [comment]
Sorry, I forgot to post an update. I tried to double the update frequency to 600000 but I still get this exception.

**Comment by robnielsen** [comment]
I am also receiving an intermittent 403 after a few hours. I'm running in eclipse with debugging turned on, and the error message returned from Netatmo is:
```
2015-07-13 06:33:32.862 [DEBUG] [.o.b.n.internal.NetatmoBinding] - Querying Netatmo API
2015-07-13 06:33:32.862 [DEBUG] [g.openhab.io.net.http.HttpUtil] - About to execute 'https://api.netatmo.net/api/getmeasure?access_token={removed}'
2015-07-13 06:33:35.693 [WARN ] [g.openhab.io.net.http.HttpUtil] - Method failed: HTTP/1.1 403 Forbidden
2015-07-13 06:33:35.693 [DEBUG] [g.openhab.io.net.http.HttpUtil] - {"error":{"code":2,"message":"Invalid access token"}}
2015-07-13 06:33:35.693 [DEBUG] [.o.b.n.internal.NetatmoBinding] - Request: MeasurementRequest[accessToken={removed}]
2015-07-13 06:33:35.693 [DEBUG] [.o.b.n.internal.NetatmoBinding] - Response: MeasurementResponse[error=NetatmoError[code=2,message=Invalid access token],status=<null>,body=<null>]
2015-07-13 06:33:35.694 [ERROR] [.o.b.n.internal.NetatmoBinding] - Invalid access token
```
I'll try and see if I can come up with a fix for this.

**Comment by robnielsen** [comment]
My above comment wasn't the issue. The issue was caused when the token expired, and the token was refreshed. The binding would make a recursive call to re-execute the binding, which would cause null pointer exceptions in the original execute call to the binding.

**Comment by teichsta** [comment]
merged #2924 into master which should fix this issue

**Comment by robnielsen** [comment]
The fix is now merged into 1.8.0

**Comment by teichsta** [comment]
one more fix merged with #2891
